### PR TITLE
Add cleaned_seq attribute to Record

### PR DESCRIPTION
**Pull request by betatim**
This adds a `cleaned_seq` attribute to `Record` which computes the cleaned version of the sequence when the record is loaded. Trying to keep output of `screed.open()` and `khmer.ReadParser()` as similar as possible.

Related to dib-lab/khmer#1591 which adds the same attribute to records (the `khmer.Read` class) that `khmer.ReadParser` creates or that is seen by `consume_fasta()`.

What tests should be added? Is calling `str(sequence)` smart? I don't know much about how the database part of screed works/assumptions it makes :-/

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [ ] `make clean diff-cover` If it introduces new functionality, is it tested?
- [ ] `make format diff_pylint_report doc` Is it well formatted?
- [ ] For substantial changes or changes to the command-line interface, is it
  documented in `CHANGELOG.md`? See [keepachangelog](http://keepachangelog.com/)
  for more details.
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
I'm -1 on this PR at the moment.

The main reason is that we've deliberately kept screed agnostic to the
question of sequence content.  So it handles RNA and protein as well as
DNA, "supports" IUPAC by not touching it, and handles genomes and genes as well
as sequencing data.  This is in stark contrast to khmer which really only is
intended to deal with DNA and sequencing data.

In particular, I bet that RNA and protein loading is going to return
non-sensical 'cleaned_seq' results :).  But, generally speaking, there are
many ways to torture input data.

The other reason (which is less of a blocker) is that when you're loading
a 3 GB genome to keep in memory, you don't want to make extra copies of each
sequence, or even preprocess it in any particular standard way.

(I was -0.5 until I wrote this all down, and now I'm solidly against it ;)

I'm not sure what the implications are for khmer and #769, but that seems like
khmer's problem, not screed's.  #1466 and #1483 are relevant.  Maybe the rule
is that khmer decorates `screed.Record` as it pleases, and `Read` and the
*decorated* `screed.Record` objects should look the same?

**Comment by ctb** [comment]
tl; dr? I think the specific details of HOW khmer expects sequences to be "cleaned" do not belong in screed, because they are not general to sequence parsing.

**Comment by betatim** [comment]
Sounds sensible, especially the point about having copies everywhere by default.

**Comment by ctb** [comment]
#wontfix. https://github.com/dib-lab/screed/pull/66#issuecomment-275124692 won the day.
